### PR TITLE
Avoid need for reindexing when adding new buchets (bug 1111893)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1111893

~~Probably needs more tests.~~

@robhudson can you shoot holes in this approach ? We discussed doing it with `missing` queries, but any reason not to do this with `must_not` ? It appears to work for me, missing features are not `True` so they are never taken into consideration.

### CODE REVIEW

File: mkt/webapps/indexers.py [hunk 1]
```diff
@@ -532,9 +527,18 @@ def get_app_filter(cls, request, additional_data=None, sq=None,
             should = [es_filter.Terms(id=list(set(app_ids)))]
             sq = sq[0:len(set(app_ids))]
 
+        # MUST NOT.
+        must_not = []
+        if data['profile'] and not no_filter:
```
Comment (robhudson) [inline mkt/webapps/indexers.py]: Seems like changing this from a double negative would make it clearer. Except `filter` is a keyword.

File: mkt/search/tests/test_views.py [hunk 2]
```diff
@@ -863,6 +863,23 @@ def test_bad_profile_on_desktop(self):
...
+        # having a value for the "new" feature.
+        mock_to_dict.return_value = {}
+        self.webapp.save()
+        self.refresh('webapp')
+        res = self.client.get(self.url, data=self.qs)
+        eq_(res.status_code, 200)
+        objects = json.loads(res.content)['objects']
+        eq_(len(objects), 1)
+        obj = objects[0]
+        eq_(obj['slug'], self.webapp.app_slug)
```
Comment (robhudson) [inline mkt/search/tests/test_views.py]: Yeah, a test on `get_app_filter` itself would be nice.

**Comment by robhudson** [comment]
I can dig it. It's simpler than the empty query so if that works I'm :+1:

File: mkt/webapps/indexers.py (hunk 1 above)
Comment (diox) [inline mkt/webapps/indexers.py]: Mmmm feed uses `filtering` query parameter (but does not store it in a variable)

Comment (diox) [inline mkt/webapps/indexers.py]: I'll keep `no_filter` for now: there are a few files passing this parameter and I don't want to increase the potential for making a mistake in that commit.

**Comment by diox** [comment]
@robhudson new version for you to look at, with more tests and a refactored `must_not` clause.

**Comment by robhudson** [comment]
Thanks for the heads up. Very nice tests. r+

**Pull request merged**